src/main/java/com/aws/greengrass/easysetup/GreengrassSetup.java

```diff
@@ -287,10 +287,13 @@ void performSetup() throws IOException, DeviceConfigurationException, URISyntaxE
             throw new RuntimeException(e);
         }
 
-        //initialize the device provisioning helper
-        this.deviceProvisioningHelper = new DeviceProvisioningHelper(awsRegion, environmentStage, this.outStream);
-
         if (needProvisioning) {
+            if (!deviceConfiguration.isDeviceConfiguredToTalkToCloud()) {
```

Comment:
this isn't right. The provisioning will perform the device configuration so that it can talk to the cloud.